File: Computação Gráfica/base_glut.py
from OpenGL.GL import *
from OpenGL.GLU import *
from OpenGL.GLUT import *
import sys
import os  # Adicionando a importação do módulo 'os'
import numpy as np

janela = 0

def InitGL():
    glClearColor(0.0, 0.0, 0.0, 0.0)
    glMatrixMode(GL_PROJECTION)  # Corrigindo para GL_PROJECTION
    glLoadIdentity()

def show_info():
    print('Empresa: ' + str(glGetString(GL_VENDOR).decode()))
    print('Modelo: ' + str(glGetString(GL_RENDERER).decode()))
    print('OpenGL: ' + str(glGetString(GL_VERSION).decode()))

def show_extensions():
    exts = glGetString(GL_EXTENSIONS).decode().split(' ')
    print('Extensoes ativas: ')
    for ext in exts:
        print(ext)

def desenha_tri():
    glBegin(GL_TRIANGLES)
    glColor3f(1.0, 0.0, 0.0)
    glVertex2f(-0.5,-0.5)
    glColor3f(0.0, 1.0, 0.0)
    glVertex2f(0.5,-0.5)
    glColor3f(0.0, 0.0, 1.0)
    glVertex2f(0,0.5)
    glEnd()

def render():
    glClear(GL_COLOR_BUFFER_BIT)
    desenha_tri()
    glutSwapBuffers()

def keyboard(key, x, y):
    k = key.decode()
    if k == 'q':
        os._exit(0)

def main():
    print('Funcao principal')
    global janela

    glutInit(sys.argv)
    glutInitDisplayMode(GLUT_DOUBLE | GLUT_RGB)
    glutInitWindowSize(500, 500)
    glutInitWindowPosition(0, 0)
    glutInitContextVersion(3, 2)
    glutInitContextProfile(GLUT_COMPATIBILITY_PROFILE)
    janela = glutCreateWindow('Janela inicial')
    glutDisplayFunc(render)
    glutKeyboardFunc(keyboard)
    InitGL()
    show_info()
    glutMainLoop()

if __name__ == '__main__':
    print("Aperte q para sair")
    main()
    

File: Computação Gráfica/base_glut.c
#include <GL/glut.h>
#include <stdlib.h>
#include <stdio.h>
#include <string.h>

int width = 500,
    height = 500,
	x=0, y=0;
 
// Declarações 
void init(void);
void render(void);
void eventos_teclado(unsigned char _key, int _x, int _y);
void show_info();
void show_extensions();

int main(int argc, char** argv) {
	glutInit(&argc, argv);
	glutInitDisplayMode(GLUT_SINGLE  | GLUT_RGB);  // Especifica o buffer a ser utilizado
	glutInitWindowSize(width,height);  // Dimensoes da Janela em pixels
	glutInitWindowPosition(0,0); // Especifica coordenada superior esquerda 
	glutCreateWindow("Janela inicial"); // Cria a janela mas só exibe após mainLoop
	init();
    show_info();
	glutDisplayFunc(render); 
	glutKeyboardFunc(eventos_teclado);
	glutMainLoop();

	return 0;
}

void show_info(){
    printf("Empresa: %s\n", glGetString(GL_VENDOR));
    printf("Modelo: %s\n", glGetString(GL_RENDERER));
    printf("OpenGL: %s\n", glGetString(GL_VERSION));
    //show_extensions();
}

void show_extensions(){
    const char* exts = glGetString(GL_EXTENSIONS);
    printf("Extensões: \n");

    char** t;
    for (t = &exts; **t != '\0'; (*t)++){
        if (**t == ' '){
            printf("\n");
        } else {
            printf("%c", **t);  
        }
    }
}

void desenha_tri(){
    glBegin(GL_TRIANGLES);
        glColor3f(1.0, 0.0, 0.0);
        glVertex2f(-0.5,-0.5);
        glColor3f(0.0, 1.0, 0.0);
        glVertex2f(0.5,-0.5);
        glColor3f(0.0, 0.0, 1.0);
        glVertex2f(0,0.5);
    glEnd();
}

void init(void){
    glClearColor(0.0, 0.0, 0.0, 0.0); // Cor do buffer

    glMatrixMode(GL_MODELVIEW);
    glLoadIdentity();
}

void render(void){
	glClear(GL_COLOR_BUFFER_BIT);   // Limpa para a cor do buffer
    desenha_tri();
	glFlush(); // Envia o desenho para o framebuffer
}

void eventos_teclado(unsigned char _key, int _x, int _y){
	switch (_key) {
		case 'q':
		case 27:
			exit(0);
			break;
	}
   glutPostRedisplay();
}